go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -572,36 +577,43 @@ func (n *OvnNode) Start(ctx context.Context) error {
 					}
 				}
 				// ensure CNI support for port binding built into OVN, as masters have been upgraded
-				if initialTopoVersion < types.OvnPortBindingTopoVersion && cniServer != nil && !isOvnUpEnabled {
-					isOvnUpEnabled, err = getOVNIfUpCheckMode()
+				if initialTopoVersion < types.OvnPortBindingTopoVersion && cniServer != nil &&
+					nc.atomicOvnUpEnabled == 0 && !config.OvnKubeNode.DisableOVNIfaceIdVer {
+					isOvnUpEnabled, err := util.GetOVNIfUpCheckMode()
 					if err != nil {
 						klog.Errorf("%v", err)
 					}
 					if isOvnUpEnabled {
 						cniServer.EnableOVNPortUpSupport()
+						atomic.StoreInt32(&nc.atomicOvnUpEnabled, 1)
 					}
 				}
 			}()
 		}
+	} else if config.OvnKubeNode.Mode == types.NodeModeDPU {
+		err = nc.updateIsOvnUpEnabled(ctx)
+		if err != nil {
+			return err
+		}
```

Comment:
> Was there a pre-existing bug in the way we handled IsOvnUpEnabled for DPU?
yes, this is the first commit of this PR. I included here because I previously included DPU support (watchPodsDPU()) for secondary network in this PR as well. The same updateIsOvnUpEnabled() function will be called by secondary network as well. But as there are some more existing bugs in that codepath, I decided to defer that into a separated PR to make this PR easier to review.

> Can't it be re-use it?
I tried that but failed as there seems a lot of code only specific to default network. it is difficult to share the code. If you have any suggestion, please let me know...

Reply:
Maybe starting with something like this (instead of `updateIsOvnUpEnabled`) 
```
if config.OvnKubeNode.Mode == types.NodeModeFull  || config.OvnKubeNode.Mode == types.NodeModeDPU {
  go func() {
    if err := upgradeController.WaitForTopologyVersion(ctx, types.OvnCurrentTopologyVersion, 30*time.Minute); err != nil {
      klog.Fatalf("Error while waiting for Topology Version to be updated: %v", err)
    }
    // upgrade complete now see what needs upgrading
    if config.OvnKubeNode.Mode == types.NodeModeFull {
      // migrate service route from ovn-k8s-mp0 to shared gw bridge
      if (initialTopoVersion < types.OvnHostToSvcOFTopoVersion && config.GatewayModeShared == config.Gateway.Mode) ||
	 (initialTopoVersion < types.OvnRoutingViaHostTopoVersion) {
	  if err := upgradeServiceRoute(bridgeName); err != nil {
            klog.Fatalf("Failed to upgrade service route for node, error: %v", err)
	  }
       }
    } 
    // ensure CNI support for port binding built into OVN, as masters have been upgraded
    if initialTopoVersion < types.OvnPortBindingTopoVersion && !isOvnUpEnabled && !config.OvnKubeNode.DisableOVNIfaceIdVer {
      isOvnUpEnabled, err := util.GetOVNIfUpCheckMode()
      if err != nil {
        klog.Errorf("%v", err)
      } else if isOvnUpEnabled {
        klog.Infof("Detected support for port binding with external IDs")
        atomic.StoreInt32(&nc.atomicOvnUpEnabled, 1)
        if cniServer != nil {
          cniServer.EnableOVNPortUpSupport()
        }
      }
    }
  }()
}
```

Reply:
@jcaamano Please take a look of the latest change. I am not happy with the change so I made the change in a separate commit. Please let me know how you think. Thanks!